experimental/tools/export-rtl/export-rtl.cpp

```diff
@@ -875,7 +879,21 @@ void VerilogWriter::writeModuleInstantiations(WriteData &data) const {
 
 void VerilogWriter::writeIOMap(hw::InstanceOp instOp, WriteData &data) const {
   IOMap ioMap(instOp, exportInfo, data.getSignalNameFunc());
-  size_t numIOLeft = ioMap.inputs.size() + ioMap.outputs.size();
+  //size_t numIOLeft = ioMap.inputs.size() + ioMap.outputs.size();
+  std::vector<std::string> totalPorts; 
+  for (auto &[modPortName, internalSignalName] : ioMap.inputs) {
+    if(std::find(totalPorts.begin(), totalPorts.end(), modPortName) != totalPorts.end()) {
+      continue;
+    }
+    totalPorts.push_back(modPortName);
+  }
+  for (auto &[modPortName, internalSignalName1] : ioMap.outputs) {
+    if(std::find(totalPorts.begin(), totalPorts.end(), modPortName) != totalPorts.end()) {
+      continue;
+    }
+    totalPorts.push_back(modPortName);
+  }
+  size_t numIOLeft = totalPorts.size();
```

Comment:
The answer to these last two comments is the same as the previous one. We need to pack multiple wires together since there is no indexing in Verilog for the module instantiation.

Reply:
> In verilog the following module instantiation is not allowed:
> 
> ```
> add addi(
> .input[0]( input_0 )
> .input[1]( input_1 )
> );
> ```
> 
> The correct version is
> 
> ```
> add addi(
> .input ( {input_1, input_0} )
> )
> ```
> 
> For this reason we need to return only the name `input` in this function. Then we have to pack the values input_1 and input_0

The reason is the same as the above reply. In Verilog-2001 standard, it only supports flattened array in module instantiation.